components/renku_data_services/users/models.py

```diff
@@ -252,12 +252,40 @@ def _to_keycloak_dict(self) -> dict[str, Any]:
         }
 
 
+@dataclass(frozen=True, eq=True, kw_only=True)
+class RenkuUser:
+    """Models a Renku user.
+
+    This models contains the username which is the slug for the user's namespace.
```

Comment:
question: I am trying to understand the purpose of this class. Maybe I am missing something but from what I can see is that it is used only as a way to make the deserialization from the users endpoints easier. Is this right?

If yes, then I think it is easier to just use `UserWithNamespace` instead of `RenkuUser` in these cases. I think even without this addition we already have too many classes/models that represent users.

Reply:
Done in a3864c58abc8e326cd5dcb302e92b1cde569fed8. I feel that this creates more code repetition though.